deepchem/transformers/__init__.py

```diff
@@ -456,3 +457,56 @@ def transform_array(self, X, y, w):
 
   def untranform(self, z):
     print("Cannot undo CoulombBinarizationTransformer.")
+
+class CDFTransformer(Transformer):
+  """Histograms the data and assigns values based on sorted list."""
+  """Acts like a Cumulative Distribution Function (CDF)."""
+  def __init__(self, transform_X=False,
+               transform_y=False,
+               bins=2,
+               dataset=None):
+    self.dataset = dataset
+    self.transform_X = transform_X
+    self.transform_y = transform_y
+    self.bins = bins
+
+  def transform(self, dataset):
+    """Performs CDF transform on data."""
+    X, y, w, ids = dataset.to_numpy()
+    w_t = w
+    ids_t = ids
+    if self.transform_X:
+      X_t = self.get_cdf_values(X)
+      y_t = y
+    if self.transform_y:
+      y_t = self.get_cdf_values(y)
+      X_t = X
+    # TODO (rbharath): Find a more elegant solution to saving the data?
+    # shutil.rmtree(dataset.data_dir)
+    # os.makedirs(dataset.data_dir)
+    # Dataset.from_numpy(dataset.data_dir, X_t, y_t, w_t, ids_t)
+    
+    return X_t, y_t, w_t, ids_t
+
+  def get_cdf_values(self, array): 
+    cols = len(array[0,:])
```

Comment:
Try `cols = np.shape[0], rows = np.shape[1]` for the more idiomatic way to do this. Also, you probably want to use `n_rows` and `n_cols` instead of `rows, columns` since the current name implies that the variable contains the actual rows/columns instead of their counts.